src/site/apt/matcher_def.apt.vm

```diff
@@ -0,0 +1,364 @@
+}
++------------------------------------------+
+
++------------------------------------------+
+public abstract class AbstractHeaderMatcher implements IHeaderMatcher {
+
+    @ConfigComponent(type = Component.Type.Parameter, desc = "The id of the matcher.")
+    private final String id;
+
+    /**
+     * Constructs the IHeaderMatcher with an id value. If {@code id} is null then a
+     * unique random id is created.
+     * 
+     * @param id the Id to use.
+     */
+    protected AbstractHeaderMatcher(String id) {
+        this.id = StringUtils.isBlank(id) ? UUID.randomUUID().toString() : id;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public String toString() {
+        return getId();
+    }
+
+    @Override
+    public Description getDescription() {
+        return DescriptionBuilder.build(this);
+    }
+}
++------------------------------------------+
+
+ So lets start by creating our matcher class and implementing the match method.
+
++------------------------------------------+
+package com.example.ratMatcher;
+
+import org.apache.rat.analysis.IHeaders;
+import org.apache.rat.analysis.matchers.AbstractHeaderMatcher;
+import org.apache.rat.config.parameters.Component;
+import org.apache.rat.config.parameters.ConfigComponent;
+
+@ConfigComponent(type = Component.Type.Matcher, name = "QSC", desc = "Reports if the 'Quality, speed and cost, pick any two' rule is violated")
+public class QSCMatcher extends AbstraactHeaderMatcher {
+
+    public QSCMatcher(String id) {
+        super(id);
+    }
+
+    @Override
+    public boolean matches(IHeaders headers) {
+        String text = headers.prune()
+        return text.contains("quality") && text.contains("speed") && text.contains("cost");
+    }
+}
++------------------------------------------+
+
+ In the above example we use the ConfigComponent annotation to identify that this is a Matcher component, that it has the name 'QSC' and a description of what it does.  
+ If the "name" was not specified it would have been extracted from the class name by removing the "Matcher" from "QSCMatcher".
+
+ The Constructor calls the AbstractHeaderMatcher with a null argument so tha the parent class will assign an id.
+
+ The matcher uses the pruned text to check for the strings.  Pruned text has only the letters and numbers from the text included and all characters are converted to 
+ lower case.  There is an issue with this matcher in that it would match the string: "The quality of Dennis Hopper's acting, as Keanu Reeves costar in 'Speed', is outstanding."
+ 
+ the correction of that is left as an exercise for the reader.  Hint: matching the pruned text can be a quick gating check for a set of more expensive regular expression checks. 
+
+** The Matcher.Builder implementation
+
+  The builder must implement the IHeaderMatcher.Builder interface.
+
++------------------------------------------+
+@FunctionalInterface
```

Comment:
same as above I'd prefer to directly link to RAT sources/interfaces/classes via Gitbox instead of copy&pasting them here.